core/src/main/scala/org/ensime/indexer/domain.scala

```diff
@@ -143,7 +255,7 @@ final case class Descriptor(params: List[DescriptorType], ret: DescriptorType) {
 
 final case class RawClassfile(
   name: ClassName,
-  generics: Option[String],
+  generics: Option[GenericClass],
```

Comment:
this is - correctly - the only place you need to add the new data structure... no need for any of the hierarchy changes above.